src/pynxtools/dataconverter/writer.py

```diff
@@ -259,8 +259,22 @@ def _put_data_into_hdf5(self):
 
         def add_units_key(dataset, path):
             units_key = f"{path}/@units"
-            if units_key in self.data.keys() and self.data[units_key] is not None:
-                dataset.attrs["units"] = self.data[units_key]
+            units = self.data.get(units_key, None)
+            if units is None:
+                return
+            if isinstance(units, pint.Unit):
+                units = str(units)
+            else:
+                try:
+                    ureg.Unit(units)
+                except pint.errors.UndefinedUnitError as exc:
+                    massage = (
+                        f"Units provided for path: '{path}@units' are not valid."
+                        f" Please provide a valid unit."
+                    )
+                    raise InvalidDictProvided(massage) from exc
```

Comment:
Shall this raise an error or simply be logged as a warning? The file can be written without this wrong unit, right?
```suggestion
                    logger.warning(message) 
```